src/poetry/console/commands/install.py

```diff
@@ -82,6 +96,24 @@ def handle(self) -> int:
             )
             return 1
 
+        if self.option("only-root") and any(
+            [self.option(key) for key in {"with", "without", "only"}]
```

Comment:
```suggestion
            self.option(key) for key in {"with", "without", "only"}
```

You should be able to drop the brackets here and just let this become a generator expression instead of a list comprehension.